Expand inference-time averaging

sirmarcel opened the pull request:
Now we can also do inversions in addition to rotations. For the case of no rotations, we just take the unit matrix.

This also fixes force predictions, which were previously not rotated back into the original direction, leading to immediate failures in dynamics.

sirmarcel committed:
Expand inference-time averaging

Now we can also do inversions in addition
to rotations. For the case of no rotations,
we just take the unit matrix.

This also fixes force predictions, which
were previously not rotated back into the
original direction, leading to immediate
failures in dynamics.

ceriottm committed:
Disable compositional contributions to avoid massive mess

ceriottm committed:
Fixed Euler angles convention for the quadrature

ceriottm committed:
On a second thought, maybe it's ZXZ

spozdn commented:
This pr doesn't compute the self-contribution energy as a quick fix. To merge it, can I ask to do a proper solution with flag, compute, or not compute self-contribution energies, @ceriottm ? 

Also, it would be nice to check if this pr provides the latest version of inference-time averaging used at the end in all the MD simulations, @sirmarcel

spozdn reviewed (changes requested):
see https://github.com/spozdn/pet/pull/11#issuecomment-2348594915

sirmarcel committed:
Make addition of self_contributions optional

sirmarcel committed:
Add option to do random augmentation for single struct calculator

sirmarcel commented:
Okay, I've made the change to make compositional contributions optional, and I've also upstreamed the option to do random augmentations from the `eqt_hacks` branch (see https://github.com/lab-cosmo/pet/compare/neighbors_convert_cpp...lab-cosmo:pet:eqt_hacks). Do you also want the changes that were made in that branch that are related to multi-GPU support for single struct calculator? I don't really see how it makes sense, given that it's built for single structures, but I can add it in.

sirmarcel committed:
Fix case of DataParallel model being loaded

Previously, this would cause problems because
the forward code expects a plain model, not
the DataParallel wrapped one.

Since we are in single inference mode here,
there is no need for parallelism and we just
unwrap the model and move on.

sirmarcel commented:
Updated to fix `DataParallel` models

spozdn committed:
Merge branch 'neighbors_convert_cpp' into neighbors_convert_cpp
